src/Clock.tsx

```diff
@@ -0,0 +1,61 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timer = 0;
+
+  componentDidMount() {
+    this.timer = window.setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(
+    { clockName: prevClockName }: Readonly<Props>,
+    { date: prevDate }: Readonly<State>,
+  ) {
+    const { clockName } = this.props;
+    const { date } = this.state;
+
+    if (prevClockName !== clockName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevClockName} to ${clockName}`);
+    }
+
+    if (prevDate !== date) {
+      // eslint-disable-next-line no-console
+      console.info(date.toLocaleTimeString());
+    }
```

Comment:
nice, but you can do it on method where you update date

Reply:
I tried, but it does not work.... at least the way I do it))